modules/age.nix

```
@@ -103,9 +116,30 @@ in
       message = "age.sshKeyPaths must be set.";
     }];
 
+    # Create a new directory full of secrets for symlinking (this helps
+    # ensure removed secrets are actually removed, or at least become
+    # invalid symlinks).
+    system.activationScripts.agenixMountSecrets = ''
+      _count="$(basename "$(readlink /run/secrets)" || echo 0)"
+      (( ++_count ))
+      echo "[agenix] symlinking new secrets generation $_count to /run/secrets..."
+      mkdir -p "${cfg.secretsMountPoint}"
+      chmod 0750 "${cfg.secretsMountPoint}"
+      mount | grep "${cfg.secretsMountPoint} type ramfs" -q || mount -t ramfs none "${cfg.secretsMountPoint}" -o nodev,nosuid,mode=0750
```

We can remove one call to `mount` here.

```suggestion
      grep "${cfg.secretsMountPoint} ramfs" -q /proc/mounts || mount -t ramfs none "${cfg.secretsMountPoint}" -o nodev,nosuid,mode=0750
```